packages/service-clients/end-to-end-tests/azure-client/package.json

```diff
@@ -101,6 +101,14 @@
 		"rimraf": "^4.4.0",
 		"typescript": "~5.1.6"
 	},
+	"fluidBuild": {
+		"tasks": {
+			"build:test": [
+				"^api-extractor:esnext",
+				"^build:esnext"
+			]
+		}
+	},
```

Comment:
Rather than this, why not update the scripts in this package? Add a build:test:esnext script and make build:test run it. Then you don't need package-specific task overrides.

Reply:
That would require additional config changes. There is nothing in task config that says test build, that doesn't have a local product build, depends on dependencies packages' product builds. Would be nice if we could comment in package.json easier.

Reply:
The package does have a local product build, though, right? `"build": "fluid-build . --task build"`. That said `build:test` indeed does not depend on `build` and thus neither on the dependencies' build output... _But_ `build:test:esm` does seem to enforce that by depending on `build:esnext` which depends on `^build:esnext`. Would an option be to change `build:test` -> `build:test:esm`, change `build` -> `build:esnext` and add a new `build` that just runs `build:esnext`?

Reply:
I realized the npm script `build` is defined but there's no tsc script, no tsconfig... so it doesn't do anything locally, just serves for the task dependency graph I guess. Which I had not seen before and I find weird. Not sure we have a good way around it, but pointing it out in case someone has a good idea.

I _think_ the proposal in the previous comments might still work? Not 100% sure.

Reply:
And then I saw https://github.com/microsoft/FluidFramework/pull/20971/files#r1594414589, so you can skip my suggestion in this thread.